db/kv_dense_channel_storage_reader.go

```diff
@@ -98,7 +99,22 @@ func (ds *DenseStorageReader) GetChanges(sinceClock base.SequenceClock, toClock
 			}
 			changedPartitions[partitionNo] = partitionChanges
 		}
-		changes = append(changes, partitionChanges.GetVbChanges(vbNo)...)
+		vbChanges := partitionChanges.GetVbChanges(vbNo)
+		changes = append(changes, vbChanges...)
+
+		if activeOnly {
+			// re-iterate over vbChanges to count the number of active
+			// use that count to determine whether we're at the limit
+			for _, logEntry := range vbChanges {
+				if !logEntry.IsRemoved() {
```

Comment:
Does IsRemoved() cover tombstones as well?  I think we should be able to identify deleted from the flags as well at this point?